### PR TITLE
create the ThirdPartyResource endpoint for service brokers

### PR DESCRIPTION
**Pull request by MHBauer**
relies on a working kubectl being set up

I am going to try and get "list broker" and "add broker" working.

### CODE REVIEW

File: k8s/service_controller/utils/utils.go [hunk 1]
```diff
@@ -51,3 +53,11 @@ func ResponseBodyToObject(r *http.Response, object interface{}) error {
 func ExtractVarFromRequest(r *http.Request, varName string) string {
 	return mux.Vars(r)[varName]
 }
+
+func KubeCreateResource(r io.Reader) (string, error) {
```
Comment (duglin) [inline k8s/service_controller/utils/utils.go]: Do me a favor... add a comment here noting that while we're using `kubectl` today, this is a temporary thing and we're going to switch to normal REST calls at some point.  Just so people know what our plans are.

Comment (MHBauer) [inline k8s/service_controller/utils/utils.go]: ya, totes.

**Comment by duglin** [comment]
LGTM assuming CI is happy

**Comment by MHBauer** [comment]
This should work, but I'd like to have a test if possible. @duglin can you show me how your new test stuff works?

**Comment by MHBauer** [comment]
Adding a node based sample service broker with:
```
curl -v -X POST -d '{"name":"test", "broker_url":"http://localhost:3000", "auth_username":"SampleAsyncServiceBrokerUser" , "auth_password":"SampleAsyncServiceBrokerPassword"}' localhost:10000/v2/service_brokers
```

Results in kubectl look like:
```
λ kubectl get ServiceBroker test -o yaml
AuthPassword: SampleAsyncServiceBrokerPassword
AuthUsername: SampleAsyncServiceBrokerUser
BrokerURL: http://localhost:3000
Created: 0
GUID: 123-123
Name: test
Updated: 0
apiVersion: cncf.org/v1alpha1
kind: ServiceBroker
metadata:
  creationTimestamp: 2016-07-22T15:11:43Z
  name: test
  namespace: default
  resourceVersion: "1873"
  selfLink: /apis/cncf.org/v1alpha1/namespaces/default/servicebrokers/test
  uid: 99a631b7-501e-11e6-8a8d-028da3bf798a
```

**Comment by duglin** [comment]
@MHBauer does the `Test_SB_Create` test cover it?

**Comment by duglin** [comment]
Of course, we'd need a kube running, right?  That's a bit harder :-)

**Comment by MHBauer** [comment]
Oh, yea. Forgot about that part. Don't you have a magic hyperkube docker incantation? If you can give this a punch and confirm that it works, let's merge it and then I'll move on to the other resources/endpoints.

**Comment by MHBauer** [comment]
@duglin how can I run that test. It seems to be complaining about starting a server. I have a k8s based server started in the background. I don't see any server starting code in the test.

I'm running:
```
λ go test -v -run Create ./tests/
Error starting server: Server already running - stop it!
exit status 1
FAIL	github.com/cncf/servicebroker/k8s/service_controller/tests	0.011s
```

**Comment by duglin** [comment]
`go test -v -run Create ./tests/` as you wrote it should work, but make sure you don't already have a service_controller running (or listening on port 10000). It will try to start one for you.

**Comment by MHBauer** [comment]
I have the k8s based one running in the background. I want to use that. I do not want it to start one for me.

**Comment by duglin** [comment]
We should be able to use/pull down `gcr.io/google_containers/hyperkube-amd64` - I'd just need to add it to the testing infrastructure to start-up.

**Comment by duglin** [comment]
hmm until we add some flag to skip the starting of the server, just comment out the call to StartServer in TestMain()

**Comment by MHBauer** [comment]
Yea, that works. Let's merge.

**Comment by MHBauer** [comment]
lgtm

File: k8s/service_controller/server/controller.go [hunk 2]
```diff
@@ -85,15 +85,16 @@ func (c *Controller) CreateServiceBroker(w http.ResponseWriter, r *http.Request)
...
+		fmt.Printf("err %v\n", err)
 		utils.WriteResponse(w, 400, err)
 		return
 	}
 
 	var catalog model.Catalog
 	err = utils.ResponseBodyToObject(resp, &catalog)
 	if err != nil {
-		fmt.Printf("Failed to unmarshal catalog: ", err)
+		fmt.Printf("Failed to unmarshal catalog: \n", err)
```
Comment (duglin) [inline k8s/service_controller/server/controller.go]: missing the %s

**Pull request merged**